src/lib/update-lock.ts

```diff
@@ -35,6 +36,29 @@ function lockFilesOnHost(appId: number, serviceName: string): string[] {
 	);
 }
 
+/**
+ * Check for rollback-{health|altboot}-breadcrumb, two files that exist while
+ * rollback-{health|altboot}.service have not exited. If these files exist,
+ * prevent reboot. If the Supervisor reboots while those services are still running,
+ * the device may become stuck in an invalid state during HUP.
+ */
+export function ensureNoHUPBreadcrumbsOnHost(): Bluebird<boolean | never> {
+	return Bluebird.all(
+		[
+			'rollback-health-breadcrumb',
+			'rollback-altboot-breadcrumb',
+		].map((filename) =>
```

Comment:
can we replace this bluebird call is [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) ?

Reply:
I think you did it since the file had bluebird all over it anyways. I don't want to push to scope of this PR but we should refactor this file to remove bluebird. Especially since the lock function is callback hell.

Reply:
I totally forgot to switch out Bluebird for Promise here. Explanation, this function was just used inside updateLock.lock before, which uses Bluebird promises because of `.catchReturn` and `.disposer` which native Promises don't implement. I've since removed ensureNoHUP... from updateLock so Bluebird promises aren';t needed anymore. Thanks for the reminder!